perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java

```diff
@@ -1567,6 +1568,98 @@ public Application approveApplication(PerunSession sess, int appId) throws Perun
+	 * @param appId
+	 * @return
+	 * @throws PrivilegeException
+	 * @throws RegistrarException
+	 */
+	public List<Group> getEmbeddedGroups(PerunSession sess, int appId) throws PrivilegeException, RegistrarException, GroupNotExistsException {
+		List<ApplicationFormItemData> appFormData = getApplicationDataById(sess, appId);
+		List<Group> embeddedGroups = new ArrayList<>();
+		String groupsPipe = null;
+		for (ApplicationFormItemData item : appFormData) {
+			if (item.getFormItem().getType() == EMBEDDED_GROUP_APPLICATION) {
+				groupsPipe = item.getValue();
+			}
+		}
+
+		if (groupsPipe != null) {
+			//Format is: "Group A#124|Group B#1212|Group C#1212"
+			String[] arr = groupsPipe.split("\\|");
+			for (String el : arr) {
+				String[] subelements = el.split("#");
+				int groupId = Integer.parseInt(subelements[subelements.length - 1]);
+				embeddedGroups.add(groupsManager.getGroupById(sess, groupId));
+			}
+		}
+
+		return embeddedGroups;
+	}
+
+	/**
+	 * Submits applications to all embedded groups
+	 * @param sess
+	 * @param groups embedded groups
+	 * @param app
+	 * @throws GroupNotEmbeddedException Group is not set as embedded in VO
+	 * @throws PerunException
+	 */
+	private void submitEmbeddedGroupApplications(PerunSession sess, List<Group> groups, Application app) throws GroupNotEmbeddedException, EmbeddedGroupApplicationSubmitionError {
+		List<Integer> notEmbeddedGroupsIds = new ArrayList<>();
+
+		for (Group group : groups) {
+
+			//skip if already member of group
+			if (groupsManager.isUserMemberOfGroup(sess, app.getUser(), group)) {
+				continue;
+			}
+
+			//check if is still embedded
+			if (groupsManager.isGroupForAutoRegistration(sess, group)) {
+				notEmbeddedGroupsIds.add(group.getId());
+			}
+		}
+
+		if (!notEmbeddedGroupsIds.isEmpty()) {
+			throw new GroupNotEmbeddedException("Group(s) with ID(s): " + notEmbeddedGroupsIds + " not embedded to VO with ID " + app.getVo().getId());
+		}
+
+		String errorMessages = "";
+
+		for (Group group : groups) {
+			try {
+				Application groupApplication = new Application();
+				groupApplication.setUser(app.getUser());
+				groupApplication.setType(AppType.EMBEDDED);
+				groupApplication.setVo(app.getVo());
+				groupApplication.setGroup(group);
+				groupApplication.setFedInfo(app.getFedInfo());
+				groupApplication.setExtSourceName(app.getExtSourceName());
+				groupApplication.setExtSourceType(app.getExtSourceType());
+				groupApplication.setExtSourceLoa(app.getExtSourceLoa());
+				groupApplication.setCreatedBy("Automatically generated");
+				groupApplication.setCreatedAt(new SimpleDateFormat("yyyy/MM/dd HH:mm:ss").format(new Date()));
+
+				try {
+					getFormForGroup(group);
+				} catch (FormNotExistsException e) {
+					createApplicationFormInGroup(sess, group);
+				}
+
+				submitApplication(sess, groupApplication, new ArrayList<>());
+			} catch (PerunException e) {
+				errorMessages = errorMessages.concat(System.lineSeparator() + e.getMessage());
```

Comment:
We should also add ids of the failed groups. I think that the best solution would be to add an extra property to the `EmbeddedGroupApplicationSubmitionError`. 
* `failedGroups` - this would be a map from `Integer` to `String`. For each failed group, it would store its id and appropriate message from the exception.

Reply:
added